scripts/convert-with-core-redcap.py

```diff
@@ -0,0 +1,58 @@
+import re
+from pathlib import Path
+
+import janitor.polars  # noqa: F401
+import polars as pl
+
+# Set the folder path for raw-data
+resource_dir = Path(__file__).resolve().parent.parent
+folder_path = resource_dir / "data-raw"
+
+# Transform from tab sep to comma sep
+with open(folder_path / "data.tsv", "r") as tsv:
+    with open(folder_path / "data-ready.csv", "w") as csv:
+        for line in tsv:
+            content = re.sub("\t", ",", line)
+            csv.write(content)
+
+# Pull in data
+df = pl.read_csv(folder_path / "data-ready.csv", infer_schema_length=100_000)
+
+block_mapping = {1: "Block 1", 2: "Block 2", 3: "Block 3"}
+
+treatment_map = {"V": "Virgin", "M": "Mating"}
+
+df = (
+    df.with_columns(
+        (pl.col("CYCLE").cast(pl.Utf8) + "0" + pl.col("ID").cast(pl.Utf8)).alias(
+            "redcap_id"
+        ),
```

Comment:
Does redcap need a unique ID or, what does redcap use the ID for or what does it need from it? If is a unique ID only necessary for Redcap, could we give the row an UUID?

Reply:
Mainly because I hadn't thought of UUIDs, and also because I wanted to experiment a bit with concatenating a couple of columns :-)

I did a bit of digging, will this issue (https://github.com/pola-rs/polars/issues/17486) create a problem if I was to switch it over to UUIDs?

Reply:
As long as you convert to `str` (https://github.com/pola-rs/polars/issues/17486#issuecomment-2213376292) it is fine :relaxed: